ecr/ref.go

```diff
@@ -70,6 +74,60 @@ func (spec ECRSpec) Registry() string {
 	return spec.arn.AccountID
 }
 
+// ParseEcrImageNameToRef takes an ECR image name and returns its canonical `ref` format.
+// i.e. ecr.aws/arn:<partition>:ecr:<region>:<account>:repository/<name>:<tag>
+func ParseEcrImageNameToRef(input string) (string, error) {
+	ref := "ecr.aws/"
+	partition := "aws"
+	if strings.HasPrefix(input, "https://") {
+		input = strings.TrimPrefix(input, "https://")
+	}
+	// Matching on account, region and TLD
+	err := errors.New("Invalid ECR image name")
+	matches := ecrRegex.FindStringSubmatch(input)
+	if len(matches) < 3 {
+		return "", err
+	}
+	tld := matches[3]
+	region := matches[2]
+	account := matches[1]
+	// If `.cn` TLD, partition should be "aws-cn"
+	// If US gov cloud regions, partition should be "aws-us-gov"
+	// If both of them match, the image source is invalid
+	isCnEndpoint := tld == ".cn"
+	isGovCloudEndpoint := region == "us-gov-west-1" || region == "us-gov-east-1"
+	if isCnEndpoint && isGovCloudEndpoint {
+		return "", err
+	} else if isCnEndpoint {
+		partition = "aws-cn"
+	} else if isGovCloudEndpoint {
+		partition = "aws-us-gov"
+	}
```

Comment:
I'd prefer that we don't hard-code this logic here.  This is incomplete (it doesn't include the `aws-iso` and `aws-iso-b` partitions) and subject to change as both new regions and new partitions get added.

The AWS SDK for Go keeps [constants for partitions and regions](https://github.com/aws/aws-sdk-go/blob/fa81b5cb4456a1e77f5ccd74d729d1761821aea7/aws/endpoints/defaults.go) and has the [`PartitionForRegion` function](https://github.com/aws/aws-sdk-go/blob/fa81b5cb4456a1e77f5ccd74d729d1761821aea7/aws/endpoints/endpoints.go#L264-L277) that can provide the correct partition given the region.